### ISSUE TITLE
Question: Coolify with Podman

### ISSUE THREAD

**Issue by matheusfenolio** [coollabsio/coolify#310]
Is it possible to run Coolify with Podman? Also, what about adding Podman as an option Destination? 

https://podman.io/

https://github.com/containers/podman

**Comment by alexanderadam** [comment]
Bonus points for _unprivileged_ Podman support :wink:

**Comment by yieme** [comment]
Podman is "the" docker replacement because it doesn't require anything to run as root. In short, a big security improvement. It was built to be a docker replacement with a simple `alias docker=podman`. It also has had a docker compatible rest api since 2020 - https://podman.io/blogs/2020/07/01/rest-versioning.html

The only thing I am aware of that might be the smallest challenge could be is someone's container using ports below 1024, and to fix that, it's just a config setting away and this I believe can be configured.

Appreciate your consideration :-)

**Comment by XPAction** [comment]
I think add this `net.ipv4.ip_unprivileged_port_start=80` to `/etc/sysctl.conf `

**Comment by alexanderadam** [comment]
I don't think privileged ports would be an issue. Since there are many solutions to it (i.e. firewall forwarding, adding capabilities, lowering the number of max privileged ports).

**Comment by andrasbacsai** [comment]
I'm closing this issue, because tracking features will be moved to GitHub discussions. Github issues will be only for bugs.

Version 4 is in full focus. v3 related issues will be closed, if it is not a critical bug.

Thanks for your understanding.

**Comment by harish2704** [comment]
Hello team,
I Just want to inform you that it is possible to run Coolify with Podman. The details are given below.

During the experiment, I found 3 things in Coolify which is not compatible with with a typical Podman setup. They are

1. Coolify has hard dependency on Docker-cli and Docker-compose plugin
  - it is not fully compatible with podman-docker or podman-compose. 
  - So I installed docker-cli from the docker repo without installing Docker service. 
  - This way, docker-cli was communicating to `podman` over `podman.sock`
  - I created a unprivileged user ( say userid=1001 ) and started podman service under that user and created symlink for `docker.sock` 
  - `ln -s /var/run/user/1001/podman/podman.sock /var/run/docker.sock`
2. Coolify is expecting a `Server` with Root access to run containers . 
  - For that, I created a simple systemd based container with `sshd` enabled on same machine ( as privileged user )
  - Bind mounted `/var/run/user/1001/podman/podman.sock` ( ie,  /var/run/docker.sock ) to that machine
  - Installed docker-cli on that machine . 
  - This way, docker in that VM is actually communicating to podman running as unprivileged user in the host machine
  - I skipped onboarding after startup and manually configured the `Server` with IP of VM and by setting up ssh key. 
3. While adding a new server, Coolify was checking the version number of docker . This is not compatible with Podman.
  - I think this behavior of Coolify should be fixed and Instead of checking `Server.version` Coolify should check `Server.ApiVersion` for compatibility 
  - I fixed this issue by manually patching ( editing ) the php source code inside the container 
  
  After that, everything was working as expected. I tried deploying couple of applications and it worked. 
  
  In short, No.3 was the  only bug like incompatibility I found during this setup.

**Comment by peaklabs-dev** [comment]
See feature request here: https://github.com/coollabsio/coolify/discussions/3137#discussioncomment-10606950
